src/wallet/wallet2.cpp

```diff
@@ -7514,7 +7601,26 @@ void wallet2::get_outs(std::vector<std::vector<tools::wallet2::get_outs_entry>>
       if (n_rct == 0)
         return rct_offsets[block_offset] ? rct_offsets[block_offset] - 1 : 0;
       MDEBUG("Picking 1/" << n_rct << " in " << (last_block_offset - first_block_offset + 1) << " blocks centered around " << block_offset + rct_start_height);
-      return first_rct + crypto::rand<uint64_t>() % n_rct;
+      
+      uint64_t pick = first_rct + crypto::rand<uint64_t>() % n_rct;
```

Comment:
It is worth pointing out that % makes the resulting random distribution non-uniform. Not sure if we care much in this case.

Reply:
Was previously like that, so should be ok.